vnext/Chakra/ChakraExecutor.cpp

```diff
@@ -203,6 +203,75 @@ struct JsRuntimeTracker
 #endif
 };
 
+#if !defined(USE_EDGEMODE_JSRT)
+
+#define IfFailRet(v) \
+    { \
+        JsErrorCode error = (v); \
+        if (error != JsNoError) \
+        { \
+            return error; \
+        } \
+    }
+
+
+JsErrorCode ChakraExecutor::RedirectConsoleToDebugger(JsValueRef debuggerConsoleObject) {
+ 
+  JsValueRef globalObject = JS_INVALID_REFERENCE;
+  IfFailRet(JsGetGlobalObject(&globalObject));
+
+  JsPropertyIdRef consolePropertyId = JS_INVALID_REFERENCE;
+  IfFailRet(JsGetPropertyIdFromName(L"console", &consolePropertyId));
+
+  JsValueRef consoleObject = JS_INVALID_REFERENCE;
+  IfFailRet(JsGetProperty(globalObject, consolePropertyId, &consoleObject));
+
+  JsValueRef undefinedValue = JS_INVALID_REFERENCE;
+  IfFailRet(JsGetUndefinedValue(&undefinedValue));
+
+  if (consoleObject == JS_INVALID_REFERENCE || consoleObject == undefinedValue)
+  {
+    return JsErrorNotImplemented;
+  }
+
+  std::string script = "";
+
+  script = script + "function patchConsoleObject$$1(global, console, debugConsole) {\n";
+  script = script + "var obj = {};\n";
+  script = script + "for (var fn in console) {\n";
+  script = script + "if (typeof console[fn] === \"function\") {\n";
+  script = script + "(function(name) {\n";
+  script = script + "obj[name] = function(...rest) {\n";
+  script = script + "console[name](rest);\n";
```

Comment:
This is terribly inefficient, not to mention annoying to read.
Can you at least use a multiline literal?
https://stackoverflow.com/questions/1135841/c-multiline-string-literal

Reply:
Sure.